javalib/src/main/scala/java/util/Date.scala

```diff
@@ -0,0 +1,40 @@
+package java.util
+
+/** Ported from Scala JS and Apache Harmony
+ * - omits deprecated methods
+ * - immutable, setTime not implemented
+ * - toString not jdk compatible
+ */
+class Date(milliseconds: Long)
+    extends Object
+    with Serializable
+    with Cloneable
+    with Comparable[Date] {
+
+  def this() = this(System.currentTimeMillis())
+
+  def after(when: Date): Boolean = milliseconds > when.getTime()
+
+  def before(when: Date): Boolean = milliseconds < when.getTime()
+
+  override def clone(): Object = new Date(milliseconds)
+
+  override def compareTo(anotherDate: Date): Int =
+    milliseconds.compareTo(anotherDate.getTime())
+
+  override def equals(obj: Any): Boolean = obj match {
+    case d: Date => d.getTime == milliseconds
+    case _       => false
+  }
+
+  def getTime(): Long = milliseconds
+
+  override def hashCode(): Int = milliseconds.hashCode()
+
+  // only implement if mutable is required
+//  def setTime(time: Long): Unit =
+//    milliseconds = time
```

Comment:
Ok, I just figured it is probably never used - in Java 1.0 the interpreter was slow with no GIT and GC was slow so a common pattern was to reuse objects to avoid GC etc. Why couldn't we change it to mutable if needed? Binary compatibility issue?